Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -258,6 +258,51 @@ extension BrowserViewController: WKNavigationDelegate {
                 self.tabManager.selectedTab?.alertShownCount = 0
                 self.tabManager.selectedTab?.blockAllAlerts = false
             }
+            
+            // Prevents synthetically activated links such as: CVE-2017-7089
+            //Follow desktop
+            //https://blog.mozilla.org/security/2017/11/27/blocking-top-level-navigations-data-urls-firefox-59
+            
+            if url.scheme == "data" {
+                do {
+                    //A click is synthetic if its value is 0 (aka WKSyntheticClickTypeNoTap).
+                    //Or if its navigationType is "other".
+                    if let clickType = navigationAction.value(forKey: "syntheticClickType") as? Int, clickType == 0 || navigationAction.navigationType == .other {
+                        /*case WebKit::WebMouseEvent::OneFingerTap:
+                            return WKSyntheticClickTypeOneFingerTap;
+                        case WebKit::WebMouseEvent::TwoFingerTap:
+                            return WKSyntheticClickTypeTwoFingerTap;
+                        }*/
+                        return decisionHandler(.cancel)
```

Comment:
Please note, the above code is done on purpose. Firefox desktop v75.0 (64-bit), MacOS blocks ALL synthetic clicks. For example:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Test Page</title>
</head>
<body>
    <p>
        <a href="data:text/html,<script>alert('hi');</script>">Should be Blocked</a>
        <a href="data:image/svg+xml;base64,eyJ0ZXN0Ijoid2hhdGV2ZXIifQo=">Should also be blocked</a>
       <a href="data:video/mp4;base64,eyJ0ZXN0Ijoid2hhdGV2ZXIifQo=">Should do something</a>
       <a href="data:,">Should render blank plain text</a>
       <a href="data:application/json,{\"test\":\"yay\"}">Should do something</a>
    </p>

    <script>
        //Synthetically change the window TopLevelDomain via window.location or window.open
        setTimeout(function(){
            window.location = "data:text/html;base64,PGh0bWw+PGJvZHkgc3R5bGU9J2JhY2tncm91bmQtY29sb3I6cmVkJz48L2JvZHk+PC9odG1sPgo=";
            //window.open("data:text/html;base64,PGh0bWw+PGJvZHkgc3R5bGU9J2JhY2tncm91bmQtY29sb3I6cmVkJz48L2JvZHk+PC9odG1sPgo=");
        }, 2000);
    </script>
</body>
</html>
```